packages/react-toast/src/components/Toast/index.tsx

```diff
@@ -49,9 +49,21 @@ export const Toast: React.FunctionComponent<
             } as ICourierMessage)
           : message;
 
-      toast(<Body {...message} icon={message?.icon ?? config?.defaultIcon} />, {
-        role: config?.role ?? "status",
-      });
+      toast(
+        <Body
+          {...message}
+          icon={
+            typeof message?.icon === "string"
```

Comment:
why can't we callback to either `message?.icon || config.defaultIcon`? This is a bit hard to read and I am not sure if this is a by-product of some product specification.

Reply:
the type made me do it like this, you are right